samcli/commands/init/interactive_init_flow.py

```diff
@@ -323,6 +325,49 @@ def _generate_from_use_case(
         )
 
 
+def _get_latest_python_runtime() -> str:
+    """
+    Returns the latest support version of Python
+    SAM CLI supports
+
+    Returns
+    -------
+    str:
+        The name of the latest Python runtime (ex. "python3.12")
+    """
+
+    # set python3.9 as fallback
+    latest_major = 3
+    latest_minor = 9
```

Comment:
Should we fail if we can't find a version? My only concern is, we now need to keep this up once a python runtime is getting deprecated. 🤔

Reply:
That's a good call out, the entire point of doing this dynamically was to avoid having to update this manually 😅 

We can fail instead of returning some kind of default.